### PR TITLE
redesigned the about window

### PR DESCRIPTION
**Pull request by aklevy**
capture d ecran 2018-07-26 a 19 19 33

### CODE REVIEW

File: base/lib/core/presenter/AboutDialog.cpp [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+                              }
+                              QListView::item:selected:active, QListView::item:selected:!active{
+                              background-color:#73C1C6;
+                              color: #1A2024;
+                              }
+                              QListView {
+                              background-color: rgb(18,23,26);
+                              margin:0px;
+                              }
+                              /* QListView::item{
```
Comment (jcelerier) [inline base/lib/core/presenter/AboutDialog.cpp]: remove this

File: base/lib/core/presenter/AboutDialog.cpp [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+                              QScrollBar::add-line:vertical {
+                              border: none;
+                              background: none;
+                              }
+                              QScrollBar::sub-line:vertical {
+                              border: none;
+                              background: none;
+                              }
+                              )_");
+  for(auto item : map)
```
Comment (jcelerier) [inline base/lib/core/presenter/AboutDialog.cpp]: `const auto&`

File: base/lib/core/presenter/AboutDialog.cpp [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  connect(softwareList,&QListWidget::currentTextChanged,this,
+          [=] (const QString& currentText)
+  {
+    license->setPlainText(map.at(currentText));
+  });
+}
+
+void AboutDialog::mousePressEvent(QMouseEvent *event)
+{
+
```
Comment (jcelerier) [inline base/lib/core/presenter/AboutDialog.cpp]: unnecessary line

File: base/lib/core/presenter/AboutDialog.cpp [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    this->setCursor(Qt::PointingHandCursor);
+  }
+  else
+  {
+    this->setCursor(Qt::ArrowCursor);
+  }
+}
+void AboutDialog::paintEvent(QPaintEvent *event)
+{
+  QPen textPen(QColor("#0092cf"));
```
Comment (jcelerier) [inline base/lib/core/presenter/AboutDialog.cpp:133]: hmmm... if these colors are reused somewhere else maybe they could go in some palette that would also be used for the splash screen for instance ? (not for today)

File: base/lib/core/presenter/AboutDialog.cpp [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+                      .arg(SCORE_CODENAME);
+
+  QString commit{GIT_COMMIT};
+
+  if (!commit.isEmpty())
+  {
+    version_text += tr("Commit: %1\n").arg(commit);
+  }
+
+  auto copyright_text = QString("Copyright © Ossia "+ QString::number(QDate::currentDate().year()));
```
Comment (jcelerier) [inline base/lib/core/presenter/AboutDialog.cpp:152]: @OSSIA/core is this the correct form for the copyright ?

File: base/lib/core/presenter/AboutDialog.hpp [hunk 6]
```diff
@@ -0,0 +1,28 @@
+#pragma once
+#include <QDialog>
+
+namespace score {
+class AboutDialog : public QDialog
```
Comment (jcelerier) [inline base/lib/core/presenter/AboutDialog.hpp]: mark it `final` (`class AboutDialog final`)

File: base/lib/core/presenter/CoreApplicationPlugin.cpp [hunk 7]
```diff
@@ -85,23 +86,8 @@ void CoreApplicationPlugin::help()
...
-  if (!commit.isEmpty())
-  {
-    version_text += tr("Commit: \n%1\n").arg(commit);
-  }
-
-  QMessageBox::about(
-      nullptr, tr("About score"),
-      tr("With love and sweat from the score team. \nVersion:\n")
-          + version_text);
+  AboutDialog aboutDialog;
```
Comment (jcelerier) [inline base/lib/core/presenter/CoreApplicationPlugin.cpp:89]: so, there's sometimes a bug with dialogs in Qt that has led to crashes in my experience ; to be safe from it instead do : 

    auto aboutDialog = new AboutDialog;
    aboutDialog->exec();
    aboutDialog->deleteLater();

**Comment by bltzr** [comment]
Do we *really* need this part :

**Comment by jcelerier** [comment]
which part @bltzr ?

**Comment by bltzr** [comment]
pastedgraphic-1
it didn't make it through email

File: base/lib/core/presenter/AboutDialog.cpp (hunk 5 above)
Comment (bltzr) [reply base/lib/core/presenter/AboutDialog.cpp:152]: well I don't really know what we should put here
and also is the license really copyright, or is it some kind of Creative Commons or whatever?

**Comment by jcelerier** [comment]
yeah legally most of the open-source libs we use require the app to show their license somewhere

**Comment by bltzr** [comment]
OK, sorry, that's the "about" window, not the splash screen ! sorry for the noise and confusion 🤦‍♂️

**Comment by bltzr** [comment]
so you can discard my previous comments - next time I'll read the title

**Comment by jcelerier** [comment]
thanks !

**Pull request merged**